artifactory/utils/readremotefile.go

```diff
@@ -0,0 +1,27 @@
+package utils
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+// Download and unmarshal a file from artifactory.
+func RemoteUnmarshal(serviceManager artifactory.ArtifactoryServicesManager, remoteFileUrl string, loadTarget interface{}) (err error) {
+	ioReaderCloser, err := serviceManager.ReadRemoteFile(remoteFileUrl)
+	if err != nil {
+		return
+	}
+	defer func() {
+		if localErr := ioReaderCloser.Close(); err == nil {
```

Comment:
See my previous comment about way we handle errors inside a defer.

Reply:
https://github.com/jfrog/jfrog-cli-core/pull/39#discussion_r549296679